frontend/crd/src/app/milestones-page/milestones/milestones.component.ts

```diff
@@ -1,23 +1,33 @@
-import {Component, OnInit} from '@angular/core';
-import {Milestone, YearLevel} from "../../../domain/Milestone";
-import {MilestoneService} from "./milestone.service";
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Milestone, YearLevel } from "../../../domain/Milestone";
+import { MilestoneService } from "./milestone.service";
+import { Subject, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'app-milestones',
   templateUrl: './milestones.component.html',
   styleUrls: ['./milestones.component.less']
 })
-export class MilestonesComponent implements OnInit {
+export class MilestonesComponent implements OnInit, OnDestroy {
+
+  private destroyed$ = new Subject<any>();
 
   constructor(
     private milestoneService: MilestoneService,
   ) {
   }
 
+  ngOnDestroy(): void {
+    this.destroyed$.next("");
+    this.destroyed$.complete();
+  }
+
   ngOnInit() {
-    this.milestoneService.getMilestones().subscribe((milestones: Milestone[]) => {
-      this.yearLevels.forEach((yearLevel) => this.milestonesMap.set(yearLevel, new Array<Milestone>()));
-      milestones.forEach((milestone) => this.milestonesMap.get(milestone.yearLevel)?.push(milestone));
+    this.milestoneService.getMilestones()
+      .pipe(takeUntil(this.destroyed$))
```

Comment:
that post was kind of old. Now you can just use takeUntilDestroyed() and don't need all the destroyed stuff

Reply:
where do you import that from? vscode doesn't recognize it and isn't giving me an option to import it from anywhere